File: app/src/main/java/com/aaronhalbert/nosurfforreddit/ExternalBrowser.java
package com.aaronhalbert.nosurfforreddit;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;

public class ExternalBrowser {
    private final Context context;

    public ExternalBrowser(Context context) {
        this.context = context;
    }

    public void launchExternalBrowser(Uri uri) {
        Intent intent = new Intent(Intent.ACTION_VIEW, uri);
        context.startActivity(intent);
    }
}

File: app/src/main/java/com/aaronhalbert/nosurfforreddit/ExternalBrowser.kt
package com.aaronhalbert.nosurfforreddit

import android.content.Context
import android.content.Intent
import android.net.Uri

class ExternalBrowser(private val context: Context) {

    fun launchExternalBrowser(uri: Uri) {
        val intent = Intent(Intent.ACTION_VIEW, uri)

        // avoid crashing if there is no supported external browser installed on the device
        intent.resolveActivity(context.packageManager)?.let { context.startActivity(intent) }
    }
}
